### ISSUE TITLE
Contract Compiler locking files

### ISSUE THREAD

**Issue by lock9** [neow3j/neow3j#1012]
**Description of the bug**
The compiler is not releasing the file lock over some files. 

![image](https://github.com/neow3j/neow3j/assets/38396062/0dac3d3a-db1d-4b77-8fb1-85e9f80ac780)

**To Reproduce**
1. Compile a contract using `./gradlew neow3j`
2. Try to delete the `build` folder
3. Alternatively, you can try to change the code and recompile it. 

**Expected behavior**
It should allow the files to be removed / updated. 

![image](https://github.com/neow3j/neow3j/assets/38396062/bb9b10b1-8f0e-45db-9291-d0475ff89ea9)

**Platform and Software**
 - OS: Windows 10 / 11
 - Software Version: 3.21.2

**Comment by csmuller** [comment]
Thanks for reporting. We're looking into it.

**Comment by gsmachado** [comment]
@lock9 thanks for the report!

Can you please try with: `./gradlew neow3jCompile` instead of `./gradlew neow3j`? Do you get the same results?

I tried to reproduce using:
- VSCode 1.85.1
- Windows 11 Home (Version 21H2, OS Build 22000.1455)
- Windows Defender: Active
- Git repository (neow3j):  https://github.com/neow3j/neow3j-boilerplate-contracts

The steps I took were simple:
1. git clone  https://github.com/neow3j/neow3j-boilerplate-contracts
2. open vscode
3. `./gradlew neow3jCompile` (successful)
4. Delete `./build` dir
5. `./gradlew newo3jCompile` (successful)
6. Changed the `HelloWorldSmartContract.java` (something simple, changed "Hello World" string to "Can I reproduce?")
7. `./gradlew neow3jCompile` (successful)

Let us know! Happy to solve this asap.

**Comment by gsmachado** [comment]
@lock9 I tried to do some more crazy things on my setup and I could reproduce it, but not reliably.

For now, while we investigate the problem, the workaround I propose is:

```powershell
./gradlew --stop
```

and then:

```powershell
./gradlew neow3jCompile
```

Or, all in a single command:

```powershell
./gradlew --stop; if ($?) { ./gradlew neow3jCompile }
```

This will make Gradle search for any running daemon and tell them to immediately stop. Then you can use Gradle again.

One of my assumptions is that the Gradle might be clashing with any instantiated from the VSCode Java extension, but it's not aware of. But this is just a hypothesis that needs confirmation with a bit more testing.

😸

**Comment by gsmachado** [comment]
@lock9 any news? Is the workaround enough for now?

**Comment by lock9** [comment]
Hi,

I didn't test it yet. The problem doesn't happen very often on my device (but happens almost all the time on the user PC).
I'll try to test this workaround today. 

>One of my assumptions is that the Gradle might be clashing with any instantiated from the VSCode Java extension, but it's not aware of. But this is just a hypothesis that needs confirmation with a bit more testing.

The problem happens even if VS Code is closed 🤔

**Comment by lock9** [comment]
Hi.

I did a few more tests, and neow3j may be the culprit.  It's not using the gradlew daemon properly.

Test:
- run `./gradlew --status`. Check the list
- run `./gradlew ` to trigger a 'normal' build. 
- Run ./gradlew --status`. The number of processes should remain the same.
- Repeat the procedure, but now instead of using just `./gradlew`, using  `./gradlew neow3j`. It will create a new process every time.

Note: It happens 50% of the time. Sometimes it works, but most times, it doesn't,

**Comment by gsmachado** [comment]
> Note: It happens 50% of the time. Sometimes it works, but most times, it doesn't,

Thanks for debugging and providing more info!

@lock9 does the workaround actually work? could you or your users make progress even with the 50% bug? If not, we will give more priority to this issue.

Please, let us know.

**Comment by lock9** [comment]
The workaround didn't work. For some reason, VSCode is hanging if I try to recompile the contract after using the stop command. It goes away and never comes back :(

People are not able to use it :(

**Comment by gsmachado** [comment]
During my tests to repro this case, this issue MIGHT BE related to https://github.com/neow3j/neow3j/issues/948 .

@mialbu @csmuller: we should talk and prioritize this one.

**Comment by lock9** [comment]
Hi @gsmachado , 

I agree with you. The problem is probably related to the gradle version (there are warnings during compilation)

**Comment by gsmachado** [comment]
We are finishing upgrading Gradle compatibility to 8.5. More about it, soon.

But, I also found this: https://github.com/gradle/gradle/issues/24082

**Comment by lock9** [comment]
>Updating to 8.0.2 appears to have resolved the issue, I'm going to give this more time and will close once I'm more confident.

I'm confident that this update will fix the issue 🙏

**Comment by csmuller** [comment]
@gsmachado Do you think this is fixed with https://github.com/neow3j/neow3j/pull/1015?

**Comment by gsmachado** [comment]
> @gsmachado Do you think this is fixed with #1015?

Yes. This issue is fixed on https://github.com/neow3j/neow3j/pull/1015 . I'm confident due to my tests. Can't reproduce this issue anymore.

Oppose to what we originally thought, **this was not** a problem with Gradle versions.
